addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1438,6 +1438,10 @@ export class OdooEditor extends EventTarget {
             && (closestBlock(start).tagName !== 'TD' && closestBlock(end).tagName !== 'TD');
         let next = nextLeaf(end, this.editable);
         const splitEndTd = closestElement(end, 'td') && end.nextSibling;
+        const previousNode = previousLeaf(start);
+        if (previousNode && previousNode.nodeType === Node.TEXT_NODE && previousNode.textContent === ' ') {
+            previousNode.textContent = '\u00A0';
```

Comment:
This feels very ad hoc. Also, transforming space to nbsp is supposed to be the responsibility of prepareUpdate + restore.

Reply:
`Range.extractContents` removes the space so `prepareUpdate` can't possibly know it wasn't intended. So it's unfortunately correct to make it unbreakable in an ad-hoc way like here IMO.
Unless @sebgeelen or @Goaman have a better idea, I'm in favor of okaying this fix.

@adsa-odoo Could you please add a test for this? The test should break before your fix and pass after. Thanks.

Reply:
@Zynton My bad, `Range.extractContents` doesn't really removes the node but it becomes invisible that even we can't see using inspect. I looked into prepareUpdate + restore and found that it detects the spererate space node but it doesn't replace the node with `&nbsp;` because it is intended that if a space node is invisible, then don't replace it with `&nbsp;` . See this, https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js#L2643C3-L2656C6

Reply:
Ah! Someone said in the chatter for the task that the browser actually removed the space. In that case this is the wrong fix indeed. `prepareUpdate` should be able to replace it with `&nbsp;`. What tests break if you remove this "disobey" code?

Reply:
Ahh ^^  it was me who said that and I was wrong, sorry for that. Four test cases break when "disobey" code is removed. The two of them are deleteBackward related and the other two are insertLineBreak and insertParagraphBreak. In each test the node is replaced with` &nbsp;`

Reply:
No problem ^^ Do you want to investigate why that is the case and try to find a fix then or would you prefer for me to take over from there?

Reply:
I investigated and this is what I found, let me explain,

`getState(...leftPos(spaceNode), DIRECTIONS.LEFT).cType & CTGROUPS.BLOCK` this statement will be true for two cases, 
i) for `cType: BLOCK_INSIDE`, and 

ii) `cType: BLOCK_OUTSIDE`. 

In our case it gets cType 8 which is `BLOCK_INSIDE` and condition will be true. Now, if we remove the whole condition than space is replaced with `&nbsp;` even spacenode is invisible. We should not do that IMO because in some cases it is necessary to keep it invisible. So do you think that is it advisable to remove the tests or we have to modify condition as a FIX such that no test case gets broken.

As a solution I have modified the condition, please have a look at the code, thanks!!

Reply:
The space should not be visible only if there is no visual effect on browser, right ? But in our case the space was visible before hitting backspace. I have written a test case for the scenario. You will have a better idea by looking into it. Thanks!